applications/OptimizationApplication/python_scripts/filtering/filter.py

```diff
@@ -0,0 +1,146 @@
+import typing
+import abc
+from importlib import import_module
+
+import KratosMultiphysics as Kratos
+from KratosMultiphysics.OptimizationApplication.utilities.union_utilities import ContainerExpressionTypes
+from KratosMultiphysics.OptimizationApplication.utilities.union_utilities import SupportedSensitivityFieldVariableTypes
+from KratosMultiphysics.OptimizationApplication.utilities.component_data_view import ComponentDataView
+
+class Filter(abc.ABC):
+    """Base filter class
+
+    This class unifies the filters used in the optimization application.
+    """
+    def __init__(self) -> None:
+        self.__component_data_view: 'typing.Optional[ComponentDataView]' = None
+
+    def SetComponentDataView(self, component_data_view: ComponentDataView) -> None:
+        """Set the component data view of the control in which this filter is being used.
+
+        This sets and allows the filter to add data/fields to space which belongs to control
+        within the optimization problem.
+
+        This component data view is not passed at the construction time because, filters are created
+        when the controls are created, hence the control data space in the optimization problem does
+        not exist until the control construction is finished.
+
+        Args:
+            component_data_view (ComponentDataView): Component data view of the control.
+        """
+        self.__component_data_view = component_data_view
+
+    def GetComponentDataView(self) -> ComponentDataView:
+        """Retrieve the component data view.
+
+        This method retrieves the component data view of the control, which can be used
+        to store data and fields.
+
+        Raises:
+            RuntimeError: If SetComponentDataView is not called first before calling this.
+
+        Returns:
+            ComponentDataView: Component data view of the control.
+        """
+        if self.__component_data_view is None:
+            raise RuntimeError("Please use SetComponentDataView first.")
+        return self.__component_data_view
+
+    @abc.abstractmethod
+    def Initialize(self) -> None:
+        """Initializes the filter.
+
+        This method initializes the filter. This is only called once in the whole optimization process.
+
+        """
+        pass
+
+    @abc.abstractmethod
+    def Check(self) -> None:
+        """Checks the filter.
+
+        This method checks the filter. This is only called once in the whole optimization process after calling the initialize.
+
+        """
+        pass
+
+    @abc.abstractmethod
+    def Finalize(self) -> None:
+        """Finalizes the filter.
+
+        This method checks the filter. This is only called once in the whole optimization process at the end.
+
+        """
+        pass
+
```

Comment:
let's also have UnfilterField which has aslo application and implemented in Helmholtz/Implicit filter.

Reply:
Agree, even in the consistent version of explicit filters can have UnfilterField. :)

Reply:
How about `UnfilteredField`, I think, there is no word as "Unfilter". :/

Reply:
done :)

Reply:
Thanks.